public/app/plugins/datasource/stackdriver/components/AnnotationQueryEditor.tsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import _ from 'lodash';
+
+import { Metrics } from './Metrics';
+import { Filter } from './Filter';
+import { AnnotationTarget } from '../types';
+import { AnnotationsHelp } from './AnnotationsHelp';
+
+export interface Props {
+  onQueryChange: (target: AnnotationTarget) => void;
+  target: AnnotationTarget;
+  datasource: any;
+}
+
+interface State extends AnnotationTarget {
+  [key: string]: any;
+}
+
+const DefaultTarget: State = {
```

Comment:
Don't get it? it's just the default state for this very component.
https://github.com/grafana/grafana/blob/d6e7fccc97b2e2acb715c2d3b738ef9f761580cd/public/app/plugins/datasource/stackdriver/components/AnnotationQueryEditor.tsx#L10
Isn't that up to the framework that handles this callback function? As of now that is the annotations_query_ctrl, but eventually it will be the framework I guess.